docker-app/qfieldcloud/core/middleware/requests.py

```
@@ -17,15 +17,20 @@ def attach_keys(get_response):
     """
 
     def middleware(request):
+        try:
+            sentry_max_size = int(config.SENTRY_REQUEST_MAX_SIZE_TO_SEND)
+        except (ValueError, TypeError):
+            logger.error(
+                "Invalid SENTRY_REQUEST_MAX_SIZE_TO_SEND value. Defaulting to 0."
+            )
+            sentry_max_size = 0
+
```

Few things things to improve there:

- The logger does not show what is the invalid value.
- renaming the config to `sentry_max_size` make the code below harder to read, you better call again `int(config.SENTRY_REQUEST_MAX_SIZE_TO_SEND)`
- setting `sentry_max_size = 0`  actually hides the error. We better want to fix it either via finding the root cause why the expected number is not a number, or just assume that somehow we get `"5"` which can be converted using the `int` function. Even if it throws again, we can then get the logs and fix the issue.

---

> * The logger does not show what is the invalid value.
changed

> * renaming the config to `sentry_max_size` make the code below harder to read, you better call again `int(config.SENTRY_REQUEST_MAX_SIZE_TO_SEND)`
ok, cool. 

> * setting `sentry_max_size = 0`  actually hides the error. We better want to fix it either via finding the root cause why the expected number is not a number, or just assume that somehow we get `"5"` which can be converted using the `int` function. Even if it throws again, we can then get the logs and fix the issue.

I guess the error is raised due to this (coming as string instead of int) or because it comes as a JSON string (`'{"__type__": "default", "__value__": aValue}'`, which means that we need to parse the JSON and extract "__value__".

---

Is this an observation or a guess? The format you mentioned is the way the data is stored in the database, see https://github.com/jazzband/django-constance/blob/22bdb011db0f57562ff14203209a08dbcb2a5194/constance/codecs.py#L46

---

It is an indirect observation as I had the same problem with another config `TRIAL_PERIOD_DAYS`.

---

The value you received for `TRIAL_PERIOD_DAYS` was in the same JSON format as above, or you don't know?

---

I did encounter an error and inspected the returned object, which was in the JSON format described above. However, this turned out to be temporary as during the next test, the expected value loaded correctly.

---

This is interesting, any way to reproduce?